themes/qgrid/source/js/menu.js

```diff
@@ -14,42 +15,214 @@ function toggleVisibility(e) {
 		overlay.style.display = 'none';
 	}
 }
-
-function filter(e) {
+function searchOnLoad() {
+	const active = document.getElementsByClassName('active-topic')[0];
+	const search = replaceWhiteSpace(location.hash.substr(8)).toLowerCase();
+	nav = document.getElementById('nav');
+	li = nav.getElementsByTagName('li');
+	if (active) {
+		if (search) {
+			pageTextSearch(search);
+			tagSearch(search);
+			menuItemsSearch(search);
+		}
+		else {
+			if (localStorage.getItem('input')) {
+				location.hash = 'search=' + localStorage.getItem('input');
+			}
+		}
+	}
+	localStorage.removeItem('input');
+}
+window.onhashchange = function () {
+	const search = replaceWhiteSpace(location.hash.substr(8)).toLowerCase();
+	if (search) {
+		pageTextSearch(search);
+	}
+}
+function filterMenu(e) {
 	e.stopImmediatePropagation();
 	const input = e.target || e.srcElement;
 	const filter = input.value.toLowerCase();
-	const nav = document.getElementById('nav');
-	const li = nav.getElementsByTagName('li');
-
+	if (filter) { location.hash = 'search=' + filter; }
+	else { location.hash = ''; }
+	const search = replaceWhiteSpace(location.hash.substr(8)).toLowerCase();
```

Comment:
instead of replaceWhiteSpace can we just have one function 
`const search = getSearch()`

in getSearch you can do whatever you want